.github/workflows/publish.yml

```diff
@@ -126,16 +123,19 @@ jobs:
           path: |
             ~/.npm
             node_modules
-          key: npm-v16-${{ runner.os }}-refs/heads/master-${{ hashFiles('package.json') }}
+          key: npm-v16-${{ runner.os }}-refs/heads/v2-${{ hashFiles('package.json') }}
 
       - name: Install Node.js and npm
         uses: actions/setup-node@v1
         with:
           node-version: 16.x
           registry-url: https://registry.npmjs.org
 
-      # Note: No need to install dependencies as we have retrieved cached `node_modules` for very
-      #       same `package.json` as stored with recent `master `build
+      - name: Install dependencies
+        if: steps.cacheNpm.outputs.cache-hit != 'true'
```

Comment:
We need to ensure `id: cacheNpm` on `Retrieve dependencies from cache`, for that to work

Reply:
great catch